Makefile

```diff
@@ -377,6 +377,9 @@ kill-trino-locally:
 install-protoc-dependencies:
 	pip install --ignore-installed protobuf==4.24.0 "grpcio-tools>=1.56.2,<2" mypy-protobuf==3.1.0
 
+install-feast-ci-locally:
+	pip install -e ".[ci]"
+
```

Comment:
does we required this 
How it differentiate with ```pip install -e ".[dev]"```

Reply:
just it is a additional target to install dev specific libraries, i had to install this on local to test and resolve one of the issue. 
https://github.com/feast-dev/feast/blob/master/setup.py#L158-L224

Reply:
@lokeshrangineni It's not mandatory of course, but you can also use `make install-python-ci-dependencies-uv` per dev [guide](https://github.com/feast-dev/feast/blob/master/docs/project/development-guide.md#feast-python-sdk--cli). Just a lot better experience imho once you set some stuff up. It also uses pinned requirements files instead of installing latest versions which means your environment is similar to where github runs the tests.

Reply:
P.S. `dev` and `ci` extras are identical now. Just decided to leave both of them not to mess with anyone's workflow

Reply:
if both are doing the same and `make install-python-ci-dependencies-uv` is recommended then better to remove this target. Updated the PR.